service/device/registry/internal/src/main/java/org/eclipse/kapua/service/device/registry/connection/option/internal/DeviceConnectionOptionServiceImpl.java

```diff
@@ -62,10 +69,23 @@ public DeviceConnectionOption update(DeviceConnectionOption deviceConnectionOpti
         //
         // Check Access
         KapuaLocator locator = KapuaLocator.getInstance();
+        DeviceConnectionService deviceConnectionService = locator.getService(DeviceConnectionService.class);
         AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
         PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
         authorizationService.checkPermission(permissionFactory.newPermission(DeviceDomains.DEVICE_CONNECTION_DOMAIN, Actions.write, deviceConnectionOptions.getScopeId()));
 
+        if (deviceConnectionOptions.getReservedUserId() != null) {
+            AndPredicateImpl deviceAndPredicate = new AndPredicateImpl();
+            deviceAndPredicate.and(new AttributePredicateImpl<>(DeviceConnectionAttributes.SCOPE_ID, deviceConnectionOptions.getScopeId()));
+            deviceAndPredicate.and(new AttributePredicateImpl<>(DeviceConnectionAttributes.RESERVED_USER_ID, deviceConnectionOptions.getReservedUserId()));
+            DeviceConnectionQuery deviceConnectionQuery = new DeviceConnectionQueryImpl(deviceConnectionOptions.getScopeId());
+            deviceConnectionQuery.setPredicate(deviceAndPredicate);
+            long cnt = deviceConnectionService.count(deviceConnectionQuery);
```

Comment:
When looking for other connections that are already assigned to this user, you have to exclude this connection itself, otherwise you will always find at least one result. Please exclude the connection you're modifying from the result set